libass/ass.c

```diff
@@ -1476,7 +1480,7 @@ char *ass_load_file(ASS_Library *library, const char *fname, FileNameSource hint
             }
         }
 
-        if (bufmax < bytes_read + CHUNK_SIZE) {
+        if (bufmax <= bytes_read) {
```

Comment:
This is always triggered for regular files: next to last read would be `bytes_read == bufmax` which triggers realloc, and then the final read return 0 and sets EOF. I think it's better to keen previous variant for this condition.